moped-editor/src/views/projects/projectView/DropDownSelectComponent.js

```diff
@@ -1,78 +1,72 @@
+const DropDownSelectComponent = ({ name, data, rowData, value, onChange }) => {
+  const phaseId = rowData?.moped_phase?.phase_id;
+  const phases = data.moped_phases;
 
-  // If lookup values is null, then it is a sub-phase list we need to generate
-  if (lookupValues === null) {
-    // First retrieve the sub-phase id's from moped_phases for that specific row
-    const allowedSubphases = props.data.moped_phases.find(
-      (item) => item?.phase_id === Number(props.rowData?.phase_id ?? 0)
-    )?.moped_subphases;
+  const subphaseOptions = useSubphaseOptions({
+    phases,
+    phaseId,
+  });
 
-    // If there are no subphases, hide the drop-down.
-    if (!allowedSubphases || allowedSubphases.length === 0) {
-      return null;
+  useEffect(() => {
+    // this effect ensures the subphase value is reset when the row's
+    // phase changes
+    if (value && !subphaseOptions.find((option) => option.value === value)) {
+      onChange("");
     }
+  }, [value, subphaseOptions, phaseId]);
 
-    // We have a usable array of sub-phase ids, generate lookup values,
-    lookupValues = allowedSubphases.reduce((obj, subphase) => {
-      obj[subphase.subphase_id] = subphase.subphase_name;
-      return obj;
-    }, {});
-  }
-
-  // empty subphases can show up as 0, this removes warning in console
-  lookupValues = { ...lookupValues, 0: "" };
+  // Hide this component if there are no related subphases
+  if (subphaseOptions.length === 0) return null;
 
-  // Proceed normally and generate the drop-down
   return (
     <FormControl>
-      <Select
-        id={props.name}
-        value={props.value ?? ""}
-        style={{ minWidth: "8em" }}
-      >
-        {Object.keys(lookupValues).map((key) => {
-          return (
-            <MenuItem
-              onChange={() => props.onChange(key)}
-              onClick={() => props.onChange(key)}
-              onKeyDown={(e) => handleKeyEvent(e)}
-              value={key}
-              key={key}
-            >
-              {lookupValues[key]}
-            </MenuItem>
-          );
-        })}
+      <Select id={name} value={value ?? ""} style={{ minWidth: "8em" }}>
+        {subphaseOptions.map(({ value, label }) => (
+          <MenuItem
+            onChange={() => onChange(value)}
+            onClick={() => onChange(value)}
+            onKeyDown={(e) => handleKeyEvent(e)}
+            value={value || ""}
+            key={value}
+          >
+            {label}
+          </MenuItem>
+        ))}
+        {/* Enable this field to be blank - will coerce to null in save callback */}
         <MenuItem
-          onChange={() => props.onChange("")}
-          onClick={() => props.onChange("")}
+          onChange={() => onChange("")}
+          onClick={() => onChange("")}
           onKeyDown={(e) => handleKeyEvent(e)}
           value=""
         >
           -
         </MenuItem>
       </Select>
-      {props.name === "phase_name" && <FormHelperText>Required</FormHelperText>}
```

Comment:
this component used to support the phase and subphase select, but this PR switch phase to an autocomplete.

i was templated to rename the component but it would nuke this diff.